liferay-echo/liferay-portal
# LPS-182033 Date values are parsed incorrectly on locales other than en_us

https://issues.liferay.com/browse/LPS-182033

This PR includes LPS-182503 to prevent conflicts, those commits are merged in the Brian master pending to sinchronize with the Liferay master

**Previous PR**: https://github.com/liferay-echo/liferay-portal/pull/11797

**Original PR comment:**

Motivation
=======
When a site's default language uses a different date format than American English a structure's date field is not correctly mapped to content pages.

Proposed Solution
========
Replaced the hardcoded en_us date parse patterns with a locale appropriate one

Steps to Verify
========
1. Create a site, set French as its default language
2. Create a structure with a date field
3. Create a web content from the structure, select the a date where the day number is higher than 12
4. Create a content page, put a paragraph fragment in it
5. Map the web content's date field into the paragraph's text
6. Publish the page

**Expected behaviour:** Displayed date should look like: "17/04/2023"
**Actual behavior:** "Displayed date gets rolled over to the next year and looks like "04/05/24"

cc: @balazssk @GezaGroma
